addons/mail/static/src/models/file_uploader.js

```diff
@@ -151,6 +152,9 @@ registerModel({
             if (activity && activity.exists()) {
                 activity.markAsDone({ attachments });
             }
+            if (channel && channel.exists()) {
+                channel.updateAvatar(attachments);
+            }
```

Comment:
Instead of naming method `updateAvatar()` with checks that put a branching to not actually update avatar (e.g. several attachments or non-image uploaded), better call it `onFileUploaded()`.

This renaming let it just be function when file is uploaded, and gives more room to do anything, like checks and/or actually update the avatar.